collections-contrib/src/main/scala/strawman/collection/decorators/HasImmutableMapOps.scala

```diff
@@ -0,0 +1,43 @@
+package strawman.collection
+package decorators
+
+/**
+  * Type class witnessing that a collection type `C` has
+  * a conversion to `immutable.MapOps[K, V, _, _]`.
+  *
+  * @see [[scala.collection.decorators.HasIterableOps]]
+  */
+trait HasImmutableMapOps[C] extends HasMapOps[C] {
+
+  // Convenient intermediate type definitions to satisfy type bounds
+  protected type _CC[X, +Y] <: immutable.MapOps[X, Y, _CC, _]
```

Comment:
If the user ever needs to interact with this type directly, I would change the name to something more memorable, e.g. `MapOf`.

Reply:
This is an implementation detail which is not supposed to be used by users. I wish I could get rid of it but couldn’t figure out how.